pkg/gitutils/gitutils.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*Package engine implements the Go text template engine as needed for Helm.
+
+When Helm renders templates it does so with additional functions and different
+modes (e.g., strict, lint mode). This package handles the helm specific
+implementation.
+*/
+package gitutils
+
+import (
+	"os"
+
+	"github.com/Masterminds/vcs"
+)
+
+func HasGitReference(gitRepo, ref, repoName string) (bool, error) {
+	local, err := os.MkdirTemp("", repoName)
+	if err != nil {
+		return false, err
+	}
+	repo, err := vcs.NewRepo(gitRepo, local)
+
+	if err != nil {
+		return false, err
+	}
+
+	if err := repo.Get(); err != nil {
```

Comment:
Is this doing a full clone just for this check? Could this be minimised into a shallow clone of the given ref, and failure means it doesn't exist? For that matter, if we only care about branches or tags (as is currently the case), [the docs imply we can get that without cloning at all](https://pkg.go.dev/github.com/Masterminds/vcs#GitRepo).